tests/TestCase/Http/MiddlewareQueueTest.php

```diff
@@ -390,4 +391,13 @@ public function testAddingDeprecatedDoublePassMiddleware(): void
             $this->assertSame($cb, $queue->current()->getCallable());
         });
     }
+
+    public function testDIContainer(): void
+    {
+        $container = new Container();
+        $container->add(SampleMiddleware::class, new SampleMiddleware());
+        $queue = new MiddlewareQueue([], $container);
+        $queue->add(SampleMiddleware::class);
+        $this->assertInstanceOf(SampleMiddleware::class, $queue->current());
```

Comment:
You need to assert it's the same instance set to the container. Instanceof check will pass even if the instance was created within the middleware queue.